pkg/disruption/backend/sampler/remote.go

```diff
@@ -0,0 +1,501 @@
+			Namespace: namespace,
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind: "ClusterRole",
+			Name: "cluster-admin",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      serviceAccountName,
+				Namespace: namespace,
+			},
+		},
+	}
+
+	client := clientset.RbacV1().ClusterRoleBindings()
+	var err error
+	if create {
+		_, err = client.Create(ctx, obj, metav1.CreateOptions{})
+	} else {
+		err = client.Delete(ctx, obj.Name, metav1.DeleteOptions{})
+	}
+
+	if apierrors.IsAlreadyExists(err) || apierrors.IsNotFound(err) {
+		return nil
+	}
+	return err
+}
+
+func callRBACHostaccess(ctx context.Context, clientset *kubernetes.Clientset, create bool) error {
+	obj := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-privileged", serviceAccountName),
+			Namespace: namespace,
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind: "ClusterRole",
+			Name: "system:openshift:scc:privileged",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      serviceAccountName,
+				Namespace: namespace,
+			},
+		},
+	}
+
+	client := clientset.RbacV1().ClusterRoleBindings()
+	var err error
+	if create {
+		_, err = client.Create(ctx, obj, metav1.CreateOptions{})
+	} else {
+		err = client.Delete(ctx, obj.Name, metav1.DeleteOptions{})
+	}
+
+	if apierrors.IsAlreadyExists(err) || apierrors.IsNotFound(err) {
+		return nil
+	}
+	return err
+}
+
+func callServiceAccount(ctx context.Context, clientset *kubernetes.Clientset, create bool) error {
+	obj := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceAccountName,
+			Namespace: namespace,
+		},
+	}
+
+	client := clientset.CoreV1().ServiceAccounts(namespace)
+	var err error
+	if create {
+		_, err = client.Create(ctx, obj, metav1.CreateOptions{})
+	} else {
+		err = client.Delete(ctx, obj.Name, metav1.DeleteOptions{})
+	}
+
+	if apierrors.IsAlreadyExists(err) || apierrors.IsNotFound(err) {
+		return nil
+	}
+	return err
+}
+
+func callProject(ctx context.Context, clientset *projectclient.Clientset, create bool) error {
+	obj := &projv1.Project{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace,
+			Labels: map[string]string{
+				"pod-security.kubernetes.io/enforce": "privileged",
+				"pod-security.kubernetes.io/audit":   "privileged",
+				"pod-security.kubernetes.io/warn":    "privileged",
+			},
+		},
+	}
+
+	client := clientset.ProjectV1().Projects()
+	var err error
+	if create {
+		_, err = client.Create(ctx, obj, metav1.CreateOptions{})
+	} else {
+		err = client.Delete(ctx, obj.Name, metav1.DeleteOptions{})
+	}
+
+	if apierrors.IsAlreadyExists(err) || apierrors.IsNotFound(err) {
+		return nil
+	}
+	return err
+}
+
+func fetchFileViaOC(ctx context.Context, nodeName string, filePath string) (string, error) {
+	args := []string{"adm", "node-logs", nodeName, fmt.Sprintf("--path=%s", filePath)}
+	cmd := exec.CommandContext(ctx, "oc", args...)
```

Comment:
no. Have a look at https://github.com/openshift/origin/blob/master/pkg/monitor/nodedetails/node_log.go#L203-L212 for how to stream a node log and read it without pulling everything into memory.